app/controllers/wpcc/engine_controller.rb

```diff
@@ -22,5 +22,37 @@ def session_information
         'eligible_salary'
       )
     end
+
+    def check_wpcc_session
+      wpcc_session_expired? ? expire_wpcc_session : update_wpcc_session_expiry
+    end
```

Comment:
Seeing the code again, I have a suggestion. 

You can remove the method above and do:

```
   before_action :expire_wpcc_session, if: :wpcc_session_expired?
   before_action :update_wpcc_session_expiry, unless: :wpcc_session_expired?
```